### ISSUE TITLE
BUG: Elision modifies non-temporary

### ISSUE THREAD

**Issue by charris** [numpy/numpy#9245]
```
In [17]: a = ones(2**20)[::2]

In [18]: b = a.flat.__array__() + 1

In [19]: del b

In [20]: a
Out[20]: array([ 2.,  2.,  2., ...,  2.,  2.,  2.])
```
The problem is that `a.flat.__array__()` is a copy of `a` that has the updateifcopy flag set. It looks like a valid temporary, but isn't. Same happens with `asarray(a.flat)` + 1`

**Comment by charris** [comment]
@juliantaylor Another problem with elision.

**Comment by charris** [comment]
@juliantaylor Do you want to fix this? I think both !updateifcopy and writeable need to be checked before using an array as a temporary. I would like to release 1.13.1 sometime next week, so will take a shot at it if you don't have the time.

**Comment by juliantaylor** [comment]
yes it is a simple fix, I'll post a PR tomorrow

**Comment by charris** [comment]
Closed by #9260. @juliantaylor A separate "closes" is needed for each issue.
